x-pack/plugin/ql/src/main/java/org/elasticsearch/xpack/ql/planner/ExpressionTranslators.java

```diff
@@ -314,6 +323,36 @@ public static Query doTranslate(Range r, TranslatorHandler handler) {
         }
     }
 
+    public static class InComparisons extends ExpressionTranslator<In> {
```

Comment:
If you're comfortable, please update the rule in SQL as well similar to `Ranges` or `BinaryLogic` - essentially extend this one from QL and add the if/else block with `onAggs`.